homeassistant/components/generic/config_flow.py

```diff
@@ -257,45 +270,46 @@ async def async_test_stream(
 
     try:
         url = yarl.URL(stream_source)
-    except ValueError:
-        return {CONF_STREAM_SOURCE: "malformed_url"}
+    except ValueError as err:
+        raise InvalidStreamException("malformed_url") from err
     if not url.is_absolute():
-        return {CONF_STREAM_SOURCE: "relative_url"}
+        raise InvalidStreamException("relative_url")
     if not url.user and not url.password:
         username = info.get(CONF_USERNAME)
         password = info.get(CONF_PASSWORD)
         if username and password:
             url = url.with_user(username).with_password(password)
             stream_source = str(url)
     try:
-        stream = create_stream(
-            hass,
-            stream_source,
-            stream_options,
-            DynamicStreamSettings(),
-            "test_stream",
+        stream = PreviewStream(
+            create_stream(
+                hass,
+                stream_source,
+                stream_options,
+                DynamicStreamSettings(),
+                "test_stream",
+            )
         )
         hls_provider = stream.add_provider(HLS_PROVIDER)
```

Comment:
this can be moved out of the `try` (i realize it was inside before)

Reply:
but `create_stream` can trigger a `HomeAssistantError` if stream is not set up.

This was added in #121308